packages/mongodb-chatbot-server/src/routes/conversations/addMessageToConversation.ts

```diff
@@ -70,6 +63,26 @@ export interface AddMessageToConversationRouteParams {
   maxInputLengthCharacters?: number;
   maxUserMessagesInConversation?: number;
   addMessageToConversationCustomData?: AddCustomDataFunc;
+  /**
+    If present, the route will create a new conversation
+    when given the `conversationIdPathParam` in the URL.
+   */
+  createConversation?: {
+    /**
+      Create a new conversation when the `conversationId` is the string "null".
+     */
+    createOnNullConversationId: boolean;
```

Comment:
Seems like there's a mismatch here between the type & the description comment. i.e. `conversationIdPathParam` vs `createOnNullConversationId`.

Based on the type, it seems like we're not implementing the more customizable route and just using `/conversations/null/messages`. If that's the case, I think we should omit `createOnNullConversationId` entirely and let the boolean check be directly on whether or not `createConversation` is defined.

When would I want to define this config and then _not_ have it enabled?

Reply:
the reason that this field exists, which is required if you specify `createConversation` is b/c there are other optional fields. 

if there weren't this `createOnNullConversationId` field, then the way that you'd configure create conversation on the endpoint w/o the optional fields would be:

```ts
{
  // ...other config
  createConversation: {/*empty obj*/}
}
```

Reply:
IMO that empty object _is_ a legit way of configuring it (i.e. presence of the field, regardless of the value)

One other small option to consider is allowing `createConversation` to be `boolean | CreateConversationConfig` and then you could do `createConversation: true` instead of the empty object.

Reply:
i'm going to keep this as is, but i understand all your points. i don't think that any of these solutions are optimal, but also think it doesn't really matter and don't want to get lost in the bikeshed here instead of delivering customer value™️